### PR TITLE
Jennyf/user credential tests

### PR DESCRIPTION
**Pull request by jennyf19**
#852 
Adding further test coverage

### CODE REVIEW

File: tests/Test.ADAL.NET.Integration/FederatedUserCredentialTests.cs [hunk 1]
```diff
@@ -280,6 +280,45 @@ public void Initialize()
...
+
+            // Call acquire token
+            AuthenticationResult result = await context.AcquireTokenAsync(TestConstants.DefaultResource, TestConstants.DefaultClientId,
+                new UserCredential(TestConstants.DefaultDisplayableId));
+
+            Assert.IsNotNull(result);
+            Assert.AreEqual("some-access-token", result.AccessToken);
+
+            // There should be one cached entry.
+            Assert.AreEqual(1, context.TokenCache.Count);
```
Comment (kpanwar) [inline tests/Test.ADAL.NET.Integration/FederatedUserCredentialTests.cs]: validate that the cache entry was updated with new access token

File: tests/Test.ADAL.NET.Integration/FederatedUserCredentialTests.cs [hunk 2]
```diff
@@ -280,6 +280,45 @@ public void Initialize()
...
+            Assert.IsNotNull(result.UserInfo);
```
Comment (kpanwar) [inline tests/Test.ADAL.NET.Integration/FederatedUserCredentialTests.cs]: check that all mocks were consumed

File: tests/Test.ADAL.NET.Integration/FederatedUserCredentialTests.cs (hunk 1 above)
Comment (jennyf19) [reply tests/Test.ADAL.NET.Integration/FederatedUserCredentialTests.cs]: done.

File: tests/Test.ADAL.NET.Integration/FederatedUserCredentialTests.cs (hunk 2 above)
Comment (jennyf19) [reply tests/Test.ADAL.NET.Integration/FederatedUserCredentialTests.cs]: done.

**Pull request merged**